%YAML:1.0

camera:
  type: "PINHOLE"

  width: 480
  height: 640

  fx: 453.2188524469703
  fy: 453.4979398200047
  cx: 238.52801838178235
  cy: 320.11656679648684

  k1: -0.05496088896628163
  k2: 0.08380675890884436
  d1: 0.0005835491157613708
  d2: -0.0012345175777400815

Tbc:
  Rbc: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [0.9998717016948464, 0.012969665609488368, -0.00940042147091448,
           0.012997302534725096, -0.9999113699284277, 0.0028848595592336464, 
           -0.009362172647073105, -0.0030066695584529905, -0.9999516536620625]
  
  tbc: !!opencv-matrix
    rows: 3
    cols: 1
    dt: d
    data: [-0.036867418714996776, -0.001624024582832584, -0.1190495925792064]

# track parameters
fm_threshold: 1.0
max_cornor_num: 150
min_cornor_gap: 30
track_frequency: 20 
track_verbose: 0

# inertial parameters
matrix_inverse: 1
inertial_init_cnt: 200
inertial_init_accl_cov: 1.0
inertial_init_gyro_cov: 0.1

# noise parameters
noise_accl: 0.0025
noise_gyro: 0.000025
noise_accl_bias: 0.0001
noise_gyro_bias: 0.000001
noise_observation: 0.0001225

# feature optimization
feature:
  max_iter_cnt: 20
  max_try_cnt: 10
  converge_threshold: 5.e-7
  min_disparity_angle: 5 # in deg
  min_disparity_distance: 0.1

# image noise
sliding_window_lens: 20
distance_threshold: 0.5
angle_threshold: 15 # in deg

# save trajectory
save_trajectory: 0
save_path: "/home/ubuntu/catkin_ws/src/learn_msckf/results/trajectory.txt"
